src/Microsoft.AspNetCore.Routing/DefaultInlineConstraintResolver.cs

```diff
@@ -82,7 +82,16 @@ private static IRouteConstraint CreateConstraint(Type constraintType, string arg
             // No arguments - call the default constructor
             if (argumentString == null)
             {
-                return (IRouteConstraint)Activator.CreateInstance(constraintType);
+                try
+                {
+                    return (IRouteConstraint)Activator.CreateInstance(constraintType);
+                }
+                catch (Exception exception)
+                {
+                    throw new InvalidRouteException(
```

Comment:
Switching away from MissingMethodException seems fine. Unlikely someone would have depended on that.